bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/MagentoGraphqlClient.java

```
@@ -0,0 +1,101 @@
+ *
+ ******************************************************************************/
+
+package com.adobe.cq.commerce.core.components.internal.models.v1;
+
+import java.lang.reflect.Type;
+import java.util.Collections;
+
+import org.apache.http.Header;
+import org.apache.http.message.BasicHeader;
+import org.apache.sling.api.resource.Resource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.adobe.cq.commerce.graphql.client.GraphqlClient;
+import com.adobe.cq.commerce.graphql.client.GraphqlRequest;
+import com.adobe.cq.commerce.graphql.client.GraphqlResponse;
+import com.adobe.cq.commerce.graphql.client.RequestOptions;
+import com.adobe.cq.commerce.magento.graphql.gson.QueryDeserializer;
+import com.day.cq.commons.inherit.ComponentInheritanceValueMap;
+import com.day.cq.commons.inherit.HierarchyNodeInheritanceValueMap;
+import com.day.cq.commons.inherit.InheritanceValueMap;
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.PageManager;
+
+/**
+ * This is a wrapper class for {@link GraphqlClient}. The constructor adapts a {@link Resource} to
+ * the GraphqlClient class and also looks for the <code>cq:magentoStore</code> property on the resource
+ * path in order to set the Magento <code>Store</code> HTTP header. This wrapper also sets the custom
+ * Magento Gson deserializer from {@link QueryDeserializer}.
+ */
+public class MagentoGraphqlClient implements GraphqlClient {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MagentoGraphqlClient.class);
+
+    public static final String STORE_CODE_PROPERTY = "cq:magentoStore";
+
+    private GraphqlClient graphqlClient;
+    private RequestOptions requestOptions;
+
+    public MagentoGraphqlClient(Resource resource) {
+        graphqlClient = resource.adaptTo(GraphqlClient.class);
+        if (graphqlClient == null) {
+            LOGGER.warn("GraphQL client not available for resource " + resource.getPath());
```

In this case this instance is invalid but the caller doesn't know about it. So they continue using it.

---

Fully agree, I left this as it was in all the existing sling models, but I agree it's much better to throw an exception. I initially changed it and then thought I leave it as it was. I will change that.